plugins/json_processor/src/json_processor/component.py

```
@@ -0,0 +1,228 @@
+                "Transformation complete: %s -> %s rows",
+                original_rows,
+                final_rows,
+            )
+
+            return Result(status="success", response=data)
+
+        except Exception as e:
+            logger.error(f"JSON transformation failed: {e!s}")
+            return Result(status="error", response=None)
+
+    def _get_input_data(self) -> pl.DataFrame | None:  # noqa: PLR0911
```

Ruff (0.12.2)

`_get_input_data` is too complex (12 > 10)

(C901)